### ISSUE TITLE
DQN.train() example

### ISSUE THREAD

**Issue by iateadonut** [DLR-RM/stable-baselines3#1582]
### ❓ Question

For DQN, I'm wondering if there is documentation/examples on how to use the train() method to train an agent after training data is created?

context:
I'm working on two agents connecting to a connect 4 server and playing a game.

Due to the nature of the program, I need both agents to connect at the same time, which requires threading.

I can loop and predict() and step() until the game is "done".

But when I try to "train()", I get an error about the logger not being set.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by iateadonut** [comment]
> Hello,
> 
> I guess you might be interested in this issue: #715

Thanks.  That might really help, but I would just like to see an example of DQN.train().

On the docs here https://stable-baselines3.readthedocs.io/en/master/modules/dqn.html

it says:
train(gradient_steps, batch_size=100)

    Sample the replay buffer and do the updates (gradient descent and update target networks)

but when I attempt to use train() after as loop of predict() and step() yields "done", but I get an error relating the the default logger not existing.  So I'm guessing I need to execute some other methods before train().  I'm just wondering what those are.

**Comment by araffin** [comment]
>  If code there is, it is minimal and working
> but when I attempt to use train() after as loop of predict() and step() yields "done", but I get an error relating the the default logger not existing. So I'm guessing I need to execute some other methods before train(). I'm just wondering what those are.

Please provide a minimal and working example to reproduce the issue, it will be easier to help you that way.
But in short, you must call `_setup_learn()` before calling `train()`, one way of achieving that is to do `model.learn(1)`.
`train()` is not really meant to be called alone.

You could also do the same as the async callback and call `setup_logger()`: https://github.com/DLR-RM/rl-baselines3-zoo/blob/master/rl_zoo3/callbacks.py#L158
